### PR TITLE
More v2 tweaks: datadollection and storage

### PR DESCRIPTION
**Pull request by Achoobert**
The datadollection.QL() never gets any data. As far as I can tell, it is a webix datacollection function, and much of that was removed from `platform`

Notable changes are to ABDataCollection.js
The full set of data wasn't getting saved before, just the IDs for the records.
Also, at no point were those IDs getting retrieved

I added a few lines to force saving the whole dataset, and force pull it into `this.dataAll`
This makes it accessible for the front end, and the app is able to load without waiting for all the data to come from the server. 
However, this doesn't seem like best practice? Please let me know the correct way to look at this

### CODE REVIEW

File: AppBuilder/platform/ABDataCollection.js [hunk 1]
```diff
@@ -51,6 +51,10 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
                data["objectName"] = this.name;
                console.error(data);
             }
+            if (data === "[Object, object]" || data === "[object Object]") {
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/ABDataCollection.js]: I wonder if you could do `if (data.toLowerCase?.() === "[object, object]")`  to catch any/all permutations of the `object, object` reference.

File: AppBuilder/platform/ABDataCollection.js [hunk 2]
```diff
@@ -701,21 +719,56 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
 
    // }
 
-   // getFirstRecord() {
+   getFirstRecord() {
+      debugger;
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/ABDataCollection.js]: `// debugger;`

File: AppBuilder/platform/ABDataCollection.js [hunk 3]
```diff
@@ -701,21 +719,56 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
...
+      return this.data;
+      // });
 
-   // 		return currItem;
-   // 	}
-   // 	else {
-   // 		return null;
-   // 	}
+      if (dc) {
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/ABDataCollection.js]: not sure if you have additional plans for this, but 
-  you have already done a `return this.data;` above and will never get here
- the name of the method is `getAllRecords()` and this only looks like you are returning 1

File: AppBuilder/platform/ABDataCollection.js [hunk 4]
```diff
@@ -701,21 +719,56 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
...
+      // return storage.get(this.refStorage()).then((data) => {
+      // data = data || {};
+      // data["reducedConditions"]["values"] = dataNew;
+      // data.bootState = this.bootState;
+      // if (data.reducedConditions?.values[0] != undefined) {
+      //    data.allValues = dataNew;
+      // }
+      var dc = this.__dataCollection;
+      console.log(dc.data.driver.getRecords());
+      debugger;
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/ABDataCollection.js]: again, assuming you are sill in process with this PR, but don't leave `debugger;` statements in live code.

File: AppBuilder/platform/ABObject.js [hunk 5]
```diff
@@ -204,6 +204,65 @@ module.exports = class ABObject extends ABObjectCore {
...
+   urlRestLog() {
+      return `/app_builder/object/${this.id}/track`;
+   }
+
+   /**
+    * @method urlRestRefresh
+    * return the url to signal a refresh for this object.
+    * @return {string}
+    */
+   urlRestRefresh() {
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/ABObject.js:253]: this one is left over from a long time ago, and I don't think it is a current `api` anymore.  You'll need to see where this is being called and make sure we rethink what that code was doing.

File: AppBuilder/platform/ABObject.js [hunk 6]
```diff
@@ -204,6 +204,65 @@ module.exports = class ABObject extends ABObjectCore {
...
+      return `/app_builder/model/refreshobject/${this.id}`;
+   }
+
+   /**
+    * @method urlCount
+    * return the url to count of data for this object.
+    * @return {string}
+    */
+   urlRestCount() {
+      return `/app_builder/model/count/${this.id}`;
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/ABObject.js]: this one is now: `/app_builder/model${this.id}/count`

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
Just a couple of things to look at.

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
Just a couple of things to look at.

File: AppBuilder/platform/ABDataCollection.js (hunk 1 above)
Comment (Achoobert) [reply AppBuilder/platform/ABDataCollection.js]: This is an error that should **only** be thrown in **development** when somone changes definitions, I saw it when a data collection was misconfigured. 
However, it was _really_ annoying to get invalid data that looked valid in the console: **should this code stay in produciton code?**

File: AppBuilder/platform/ABDataCollection.js (hunk 3 above)
Comment (Achoobert) [reply AppBuilder/platform/ABDataCollection.js]: this.data is an array, it should just be dumping the whole thing back

File: AppBuilder/platform/ABObject.js (hunk 5 above)
Comment (Achoobert) [reply AppBuilder/platform/ABObject.js:253]: Added notify.developer

**Pull request merged**